pkg/agent/config/config.go

```diff
@@ -359,14 +359,6 @@ func get(ctx context.Context, envInfo *cmds.Agent, proxy proxy.Proxy) (*config.N
 		return nil, errors.Wrap(err, "failed to retrieve configuration from server")
 	}
 
-	// If the supervisor and externally-facing apiserver are not on the same port, tell the proxy where to find the apiserver.
-	if controlConfig.SupervisorPort != controlConfig.HTTPSPort {
-		isIPv6 := utilsnet.IsIPv6(net.ParseIP([]string{envInfo.NodeIP.String()}[0]))
-		if err := proxy.SetAPIServerPort(ctx, controlConfig.HTTPSPort, isIPv6); err != nil {
```

Comment:
Had to move this out of this function because get() is called with a context that is cancelled once the config has been retrieved from the server, due to it being handled in a loop. We needed to move it out to a call site with a more stable context.